src/crypto/rand/rand_stm32wle5.go

```diff
@@ -0,0 +1,65 @@
+//go:build stm32wle5
+// +build stm32wle5
+
+package rand
+
+import (
+	"device/stm32"
+	"errors"
+)
+
+func init() {
+	Reader = &reader{}
+}
+
+type reader struct {
+}
+
+const maxRngWaitTimeout = 1000
+
+var errRandom = errors.New("STM32WL TRNG timeout")
+
+func (r *reader) Read(b []byte) (n int, err error) {
+	if len(b) == 0 {
+		return
+	}
+
+	var randomByte uint32
+	for i := range b {
+		// Call rand_s every four bytes because it's a C int (always 32-bit in
+		// Windows).
+		if i%4 == 0 {
+			randomByte, err = getRand32()
+			if err != nil {
+				return n, err
+			}
+		} else {
+			randomByte >>= 8
+		}
+		b[i] = byte(randomByte)
+	}
+
+	return len(b), nil
+}
+
+func getRand32() (uint32, error) {
+
+	// Enable TRNG clock and peripheral
+	stm32.RCC.AHB3ENR.SetBits(stm32.RCC_AHB3ENR_RNGEN)
+	if !stm32.RNG.CR.HasBits(stm32.RNG_CR_RNGEN) {
+		stm32.RNG.CR.SetBits(stm32.RNG_CR_RNGEN)
+	}
```

Comment:
Do you need to do for every call or this is something that can be done once, for example in the init function?

Reply:
Most of the code looks very portable to most stm32.  Perhaps put just the initialization code that references AHB3 in a chip-specific file, and the other code for all stm32?   @ofauchon if you don’t fancy doing that, I can take a pass after this is merged.